hw_diag/utilities/security.py

```diff
@@ -0,0 +1,37 @@
+from os.path import dirname, abspath, join
+
+import gnupg
+from hm_pyhelper.logger import get_logger
+
+gpg = gnupg.GPG()
+LOGGER = get_logger(__name__)
+KEY_PATH = "keys/key-file-pub.gpg"
+
+
+def import_key():
+    root_path = dirname(dirname(dirname(abspath(__file__))))
+    full_key_path = join(root_path, KEY_PATH)
+    with open(full_key_path, "r") as f:
+        import_result = gpg.import_keys(f.read())
+        if import_result.results:
+            LOGGER.debug(f'Importing {KEY_PATH} succeed.')
+            return True
+
+    LOGGER.debug(f'Importing {KEY_PATH} failed.')
+    return None
+
+
+# Decrypt file using the gnupg
+# Return values:
+#   Decrypted data if successful
+#   None if decryption is unsuccessful
+def decrypt_file(file):
```

Comment:
Should be verification, not decryption: https://docs.red-dove.com/python-gnupg/#verification

Reply:
Make sense.

Reply:
I was wrong about this. There is no way to extract the relevant payload without using decrypt.

Reply:
Now, updated to use `decrypt()` method but with a semantic name.

Reply:
> without using

[get_payload_from_clearsigned_message](https://github.com/NebraLtd/hm-pyhelper/blob/master/hm_pyhelper/util/pgp.py#L4) would beg to differ :-) But using decrypt here is the better (and more reliable) choice since GPG and python-gpg are already dependencies for hm-diag.